libc++ on FreeBSD always uses PathScale libcxxrt and cannot change to
LLVM libc++abi without breaking backward compatibility. Besides, mixing
different C++ ABIs is not supported unless subset via DT_FILTER.

https://github.com/llvm/llvm-project/commit/35479ffb1251
https://github.com/freebsd/freebsd-src/commit/cf56074e5271

--- gcc/cp/g++spec.cc.orig	2022-07-31 22:32:16 UTC
+++ gcc/cp/g++spec.cc
@@ -60,7 +60,7 @@ along with GCC; see the file COPYING3.  If not see
 #endif
 
 #ifndef LIBCXXABI
-#define LIBCXXABI "c++abi"
+#define LIBCXXABI "cxxrt"
 #endif
 #ifndef LIBCXXABI_PROFILE
 #define LIBCXXABI_PROFILE LIBCXXABI
